pkg/gardenlet/controller/shoot/shoot_control.go

```diff
@@ -389,22 +503,42 @@ func (c *Controller) reconcileShoot(shoot *gardencorev1beta1.Shoot, logger *logr
 		}
 	}
 
-	if err := c.runReconcileShootFlow(o, operationType); err != nil {
+	if err := c.runReconcileShootFlow(o); err != nil {
```

Comment:
If there is an error during reconcile, it will return an error ->  leads to another reconcile -> syncs the Cluster resource containing the Shoot to the Seed. Is that the reason for not syncing the Cluster resource to the Seed here?

Reply:
Hm, you are asking because we are syncing the `Cluster` resource again when the reconciliation is successful but not when it is errorneous, right? Yes, basically, it would be more consistent to sync it here as well. @timuthy WDYT?

Reply:
Good point but I wouldn't probably do it, at least not just for consistency reasons. The cluster resource is a rather big resource and an extra sync does not seem beneficial.